File: filesystem_recursive_directory_listing.py
# Recursive directory listing

import harfang as hg

def entry_type_to_string(type):
	return {hg.DE_File: 'file', hg.DE_Dir: 'directory', hg.DE_Link: 'link'}[type]

entries = hg.ListDirRecursive('resources', hg.DE_All)

for entry in entries:
	print('- %s is a %s' % (entry.name, entry_type_to_string(entry.type)))

File: filesystem_recursive_directory_listing.lua
-- Recursive directory listing

hg = require("harfang")

function entry_type_to_string(type)
	tp={}
	tp[hg.DE_File], tp[hg.DE_Dir], tp[hg.DE_Link] = 'file', 'directory', 'link'
	return tp[type]
end

entries = hg.ListDirRecursive('resources', hg.DE_All)

for i=0, entries:size() do
	entry = entries:at(i)
	print(string.format('- %s is a %s', entry.name, entry_type_to_string(entry.type)))
end